tools/codegen/api/types.py

```diff
@@ -240,6 +242,47 @@ class DispatcherArgument:
     def __str__(self) -> str:
         return f"{self.type} {self.name}"
 
+@dataclass(frozen=True)
+class DispatcherSignature:
```

Comment:
Can you add a class comment describing what the dispatcher signature is for and how it differs from NativeSignature / CppSignature ? Same for those classes.

Side note: maybe rename CppSignature to ApiSignature or FrontendSignature since they're all kind of Cpp.

Reply:
This explanation would overlap with the top level comments in `tools.codegen.api.FOO` modules, so I'd prefer not to do it here. Maybe a Note reference would be better?

Wondering maybe we can reorganize these modules a bit so that the types aren't splattered about. One possibility is to get rid of types module and move the type definitions to their respective core modules. That will result in cycles when we reference types from the other modules. We could maybe resolve these cycles using the trick in https://mypy.readthedocs.io/en/stable/common_issues.html#import-cycles This looks kind of ugly so I tried not to do it this way, but maybe it is worth it.

> Side note: maybe rename CppSignature to ApiSignature or FrontendSignature since they're all kind of Cpp.

Vetoing api because then we'll get `tools.codegen.api.api` (I guess we could rename the first `api` in this list, though I'm not sure what too; `cconv` as in calling convention?) Frontend might be OK; there's some potential for confusion with Python frontend but I guess it's not too bad.